File: codes/Constructeurs/src/Vehicule.java
public class Vehicule {

    public String marque = null;
    public Moteur m = null;
    /* Constructeur par défaut */
    public Vehicule() { this.marque = "Peugeot"; this.m = new Moteur(); }
    /* Constructeur sympa */
    public Vehicule(String marque) {
        this.marque = marque;
        this.m = new Moteur();
    }
    /* Constructeur par recopie */
    public Vehicule(Vehicule v) {
        // this.moteur = v.moteur; // Hérésie
        this.m = new Moteur(); // Pas top, mais mieux
        this.marque = v.marque; // Possible et toléré car String immutable
        this.marque = new String(v.marque);
    }
}

File: codes/Constructeurs/vehicule.py
class Vehicule:

    def __init__(self):
        self.marque = "Peugeot"

    def __init__(self, marque): # Vient d'écraser le constructeur précédent
        self.marque = marque

# v = Vehicule() # Plante ! Pas de surcharge de constructeur !
m2 = "Toyota"
v2 = Vehicule(m2)
m3 = "Nissan"
v3 = Vehicule(m3)
print("v2.marque: " + v2.marque)
print("v3.marque: " + v3.marque)
